### ISSUE TITLE
Feature Request: workspace.addObjects() outputs an error on invalid input

### ISSUE THREAD

**Issue by mldichter** [NREL/openstudio-standards#540]
Currently the workspace.addObjects() function will either add all the objects in the input to the idf or none of the objects in the input to the idf and the ruby code will continue. This makes debugging code difficult as no error is generated and no information on which element of the input caused the error is outputted. You can only tell because the idf fill will be missing objects being added by the measure.

https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-2.0.3-doc/utilities/html/classopenstudio_1_1_workspace.html#a1ce26a7d2ad612bc95782e1fb43fb9d7

Please change the behavior of the workspace.addObjects() function to output information useful for debugging invalid input.

**Comment by mldichter** [comment]
>> Does the return value return an empty vector in the case that no objects were added?
Yes. That is exactly my difficulty.

Rather than trying to use openstudio, I'm going to alter my measure to modify the idf text files directly and create energyplus run group files instead of using PAT. This issue, combined with the idd file not easily changed for openstudio and "transition responsibility for management, development, distribution, and support of the OpenStudio Application to third parties by April 2020", making my own less capable but simpler tool seems like the best option.

I'm also in over my head on a lot of the coding syntax used in the measures. For instance, I don't really know what boost optional is still, which shows up as errors sometimes. We HVAC engineers after all, not computer scientists.

Thanks for you help.
    -Mitchal

**Comment by mdahlhausen** [comment]
This is an OpenStudio issue, not an openstudio-standards issue.  It doesn't seem to need a fix: just check whether the workspace.addObjects() returns false. If it does, it had an invalid argument.

**Comment by mldichter** [comment]
Thanks for letting me know about the OpenStudio issue versus openstudio-standards issue difference. I've submitted several issues that don't belong in the openstudio-standards issue. My apologies. Is there a guide for which issues should be submitted to which location, and how do I move the issues from one list to the other?

You are technically correct, the best kind of correct! Changing the behavior of the workspace.addObjects() is a feature request and not a bug. The function does work. However, it has caused me much angst as I've tried to add many different objects at once in multiple programs, and currently workspace.addObjects()'s false output tells me there is at least one problem with at least one of my objects. This problem is also chronic as each energyplus version upgrade seems to change the fields of a few objects, so I have to go back through my code and try to figure out what changed, with only an output of false to go on. See the following unmethours posts for examples.

https://unmethours.com/question/35742/debugging-of-workspaceaddobjects-input/

https://unmethours.com/question/32993/force-measure-to-assign-string-to-field-whether-or-not-the-string-is-valid-or-invalid/

https://unmethours.com/question/32179/inject-idf-objects-measure-not-adding-objects/
